packages/hash/api/src/graphql/typeDefs/knowledge/entity.typedef.ts

```diff
@@ -0,0 +1,95 @@
+import { gql } from "apollo-server-express";
+
+export const knowledgeEntityTypedef = gql`
+  interface KnowledgeEntity {
+    # These fields are repeated everywhere they're used because
+    # (a) GQL requires it - https://github.com/graphql/graphql-spec/issues/533
+    # (b) string interpolation breaks the code generator's introspection
+    #
+    # Could maybe use a custom schema loader to parse it ourselves:
+    # https://www.graphql-code-generator.com/docs/getting-started/schema-field#custom-schema-loader
+    #
+    # For now, _COPY ANY CHANGES_ from here to any type that 'implements Entity'
+    """
+    The id of the entity
+    """
+    entityId: ID!
+    """
+    The specific version of the entity
+    """
+    entityVersionId: String!
+    """
+    The id of the account that owns this entity.
+    """
+    ownedById: ID!
+    """
+    The fixed id of the type this entity is of.
+    """
+    entityTypeId: ID!
+    """
+    The full entity type definition.
+    """
+    entityType: EntityType
+    """
+    The linked entities of the entity.
+    """
+    linkedEntities: [KnowledgeEntity!]!
+    """
+    The JSON object containing the entity's properties.
+    """
+    properties: JSONObject!
+  }
+
+  """
+  For referring to an existing entity owned by a specific account id
+  """
+  input KnowledgeExistingEntity {
+    """
+    This may be a reference to a placeholder set using placeholderId on a previous UpdatePageContentsAction.
+    """
+    entityId: ID!
+    ownedById: ID!
+  }
+
+  """
+  Select entity types by ONE of componentId, entityTypeVersionedUri
+  """
+  input KnowledgeEntityTypeChoice {
+    """
+    For entity types related to block types, the URI of the block. 'componentId' in the entity type's schema.
+    """
+    componentId: ID
+    """
+    A fixed entity type ID. This may be a reference to a placeholder set using a previous createEntityTypeAction.
+    """
+    entityTypeVersionedUri: String
```

Comment:
What did we decide to call these URIs? `VersionedUri`s or just `Id`?

Reply:
I don't think this has been properly decided. The query API uses `versionedUri` in some places, but currently the `PersistedOntologyIdentifier` in the Graph API specifies `uri` which is ambiguous.

cc @TimDiekmann @Alfred-Mountfield

Reply:
I think last time we spoke about this we had basically decided to use `Id` wherever we're specifically talking about the identifier for the type. So, `EntityTypeId`, etc. and we use `baseUri` wherever it's needed. Outside of that, there are probably very few places we want to be using `uri`.

Reply:
Got it, I'll change it to `entityTypeId`. We probably want to change that in our `Persisted*TypeInterface` GQL interfaces/types as well. Might be good to do before Nate starts making use of the new BP hooks